.github/workflows/cpp-packaging.yml

```diff
@@ -39,26 +37,9 @@ env:
   xcodeVersion: "12"
 
 jobs:
-  prepare_matrix:
-    runs-on: ubuntu-latest
-    env:
-      USE_EXPANDED_MATRIX: 0
-    outputs:
-      use_expanded_matrix: ${{ steps.export-result.outputs.use_expanded_matrix }}
-    steps:
-      - name: Use expanded matrix
-        if: github.event_name == 'schedule' || github.event.inputs.use_expanded_matrix == '1'
-        run: |
-          echo "USE_EXPANDED_MATRIX=1" >> $GITHUB_ENV
-      - id: export-result
-        run: |
-          echo "USE_EXPANDED_MATRIX: ${USE_EXPANDED_MATRIX}"
-          echo "::set-output name=use_expanded_matrix::${USE_EXPANDED_MATRIX}"
-
```

Comment:
Exporting the configuration into a job output us to normalize the inputs across all worfklow triggers.  Additionally duplicate checks like this:

`if github.event.inputs.verboseBuild != 0 && github.event.inputs.verboseBuild != "" `

become simply 

`if ${{ needs.prepare_matrix.outputs.verboseBuild }}`

across all of our workflows. 

The two prepare_matrix and log_inputs could be combined.

Reply:
I agree, but I'd like to do this cleanup in a followup.

Reply:
ok.